text/0000-global-allocators.md

```diff
@@ -240,9 +252,47 @@ The allocator APIs could be simplified to a more "traditional"
 malloc/calloc/free API at the cost of an efficiency loss when using allocators
 with more powerful APIs.
 
+The global allocator could be an instance of the `Allocator` trait. Since that
+trait's methods take `&mut self`, things are a bit complicated however. The
+allocator would most likely need to be a `const` type implementing `Allocator`
+since it wouldn't be sound to interact with a static. This may cause confusion
```

Comment:
Oh, yeah, I totally overlooked the `&mut self` issue. We could side-step this by changing the trait (I'm not a fan of that, for reasons I'll outline separately) or by changing how the allocator is accessed. By the latter I mean, for example, tagging a `static X: MyAllocator` as the global allocator creates an implicit `const X_REF: &MyAllocator = &X;` and all allocation calls get routed through that. This feels extremely hacky, though, and brings back the aforementioned identity confusion.